spec.html

```diff
@@ -28634,7 +28634,7 @@ <h1>String.prototype.replace ( _searchValue_, _replaceValue_ )</h1>
             1. Let _replStr_ be ? ToString(_replValue_).
           1. Else,
             1. Let _captures_ be a new empty List.
-            1. Let _replStr_ be GetSubstitution(_matched_, _string_, _pos_, _captures_, *undefined*, _replaceValue_).
+            1. Let _replStr_ be ? GetSubstitution(_matched_, _string_, _pos_, _captures_, *undefined*, _replaceValue_).
```

Comment:
This `?` was necessary before this change, but after this change it looks like `GetSubstitution` can no longer return an abrupt completion. If that's right, this should be `!` (or omitted).

Reply:
That was my first thought too, but please see steps for `$<` in [Replacement Text Symbol Substitutions table](https://tc39.es/ecma262/#table-45): getting named group value may turn abrupt completion.

//cc @ljharb

Reply:
Well, that's not good.

I think that table should be done away with. The right-hand column mixes code points, aliases, declarative prose descriptions of text, and imperative algorithms steps which can return abrupt completions. Probably it is time to just make a big if-else.

Reply:
whoops. I'll convert these back to `?` in a followup commit, and then we can do the larger change via PR.

Reply:
Fixed in 37728d736e9e6a1e2ba951dbeb3fc096914193d6

Reply:
37728d7 is over-zealous: when `namedCaputres` is `*undefined*`, the operation genuinely cannot throw. So the first invocation of `GetSubstitution` should remain `!`.

Reply:
Perhaps #1732 resolves some of those concerns?

Reply:
8fd2e014fa52a1c932686904a7b46eb742fe3112

Reply:
#1732 is definitely a step in the right direction, and any further changes to this section would have conflicts with that PR, so that's probably the right place for further discussion.